solutions/fully-configurable/main.tf

```diff
@@ -1,7 +1,40 @@
+
+  kms_instance_crn = var.existing_kms_instance_crn != null ? var.existing_kms_instance_crn : var.existing_kms_key_crn != null ? "${join(":", slice(local.crn_parts, 0, length(local.crn_parts) - 2))}::" : null
+  kms_key_crn      = var.enable_kms_encryption ? (var.existing_kms_key_crn != null ? var.existing_kms_key_crn : module.kms[0].keys[format("%s.%s", local.kms_key_ring_name, local.kms_key_name)].crn) : null
+
+  kms_region        = var.enable_kms_encryption && (var.existing_kms_instance_crn != null || var.existing_kms_key_crn != null) ? module.existing_kms_crn_parser[0].region : null
+  kms_service_name  = var.enable_kms_encryption && (var.existing_kms_instance_crn != null || var.existing_kms_key_crn != null) ? module.existing_kms_crn_parser[0].service_name : null
+  kms_instance_guid = var.enable_kms_encryption && (var.existing_kms_instance_crn != null || var.existing_kms_key_crn != null) ? module.existing_kms_crn_parser[0].service_instance : null
+  kms_account_id    = var.enable_kms_encryption && (var.existing_kms_instance_crn != null || var.existing_kms_key_crn != null) ? module.existing_kms_crn_parser[0].account_id : null
+
+  kms_key_ring_name = var.app_config_key_ring_name != null ? "${local.prefix}${var.app_config_key_ring_name}" : null
+  kms_key_name      = var.app_config_key_name != null ? "${local.prefix}${var.app_config_key_name}" : null
+
+  create_kms_cross_account_auth_policy = var.skip_app_config_kms_iam_auth_policy && var.ibmcloud_kms_api_key != null
+
+  en_region = var.enable_event_notification && var.existing_event_notifications_instance_crn != null ? module.existing_en_crn_parser[0].region : null
+
+  kms_endpoint = var.enable_kms_encryption && (var.existing_kms_instance_crn != null || var.existing_kms_key_crn != null) ? (
+    local.kms_service_name == "hs-crypto" ? (
+      var.kms_endpoint_type == "private" ? "https://${local.kms_instance_guid}.api.private.${local.kms_region}.hs-crypto.appdomain.cloud" : "https://${local.kms_instance_guid}.api.${local.kms_region}.hs-crypto.appdomain.cloud"
+      ) : (
+      var.kms_endpoint_type == "private" ? "https://${local.kms_region}.${local.kms_service_name}.cloud.ibm.com" : "https://${local.kms_region}.${local.kms_service_name}.cloud.ibm.com"
+    )
+  ) : null
+
+  en_endpoint = var.enable_event_notification && var.existing_event_notifications_instance_crn != null ? var.event_notifications_endpoint_type == "private" ? "https://private.${local.en_region}.event-notifications.cloud.ibm.com" : "https://${local.en_region}.event-notifications.cloud.ibm.com" : null
+
+  kms_key_id = var.enable_kms_encryption && var.existing_kms_key_crn != null ? module.kms_key_crn_parser[0].resource : (var.existing_kms_instance_crn != null && var.enable_kms_encryption ? module.kms[0].keys[format("%s.%s", local.kms_key_ring_name, local.kms_key_name)].key_id : null)
 }
 
+data "ibm_iam_account_settings" "iam_account_settings" {
+  count = local.create_kms_cross_account_auth_policy ? 1 : 0
+}
```

Comment:
We should avoid using `ibm_iam_account_settings` when possible. It requires you to be an account admin. You should be able to get the account ID from the App config module output `app_config_account_id`

Reply:
Ohh so this is a special case for cross account auth policy is it?